### PR TITLE
UCT/DC: Use inline arrays for DCIs.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2430/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2431/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1776/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1777/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2432/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1778/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alex-mikheev can you pls review?

File: src/uct/ib/dc/base/dc_ep.c [hunk 1]
```diff
@@ -50,11 +50,16 @@ ucs_status_t uct_dc_ep_pending_add(uct_ep_h tl_ep, uct_pending_req_t *r)
      * - dci is either assigned or can be assigned
      * - dci has resources
      */
-    if (uct_rc_iface_has_tx_resources(&iface->super) &&
-        ((ep->dci != UCT_DC_EP_NO_DCI) || uct_dc_iface_dci_can_alloc(iface)) &&
-        uct_dc_iface_dci_ep_can_send(ep))
-    {
-        return UCS_ERR_BUSY;
+    if (uct_rc_iface_has_tx_resources(&iface->super)) {
```
Comment (alex-mikheev) [inline src/uct/ib/dc/base/dc_ep.c]: are you sure this is the same condition as above ?

File: src/uct/ib/dc/base/dc_iface.c [hunk 2]
```diff
@@ -160,38 +153,44 @@ UCS_CLASS_INIT_FUNC(uct_dc_iface_t, uct_rc_iface_ops_t *ops, uct_md_h md,
     UCS_CLASS_CALL_SUPER_INIT(uct_rc_iface_t, ops, md, worker, dev_name, rx_headroom,
                               rx_priv_len, &config->super.super);
 
-    self->tx.ndci   = config->ndci;
-    self->tx.policy = config->tx_policy;
-    ucs_debug("using %s dci selection algorithm",
-              uct_dc_tx_policy_names[self->tx.policy]);
+    if (config->ndci <= 1) {
+        ucs_error("dc interface must have at least 1 dci");
```
Comment (alex-mikheev) [inline src/uct/ib/dc/base/dc_iface.c]: < 1

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2437/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1780/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2443/ for details.

**Pull request merged**